### ISSUE TITLE
Odd completion behavior within comments

### ISSUE THREAD

**Issue by efroemling** [python-lsp/python-lsp-server#411]
I recently upgraded to Emacs 29 and got eglot wired up with python-lsp-server within python-ts-mode for my project. Everything is mostly super awesome compared to my old pure-flycheck setup. There's just one bit of odd behavior I'm seeing that I wanted to ask about:

Whenever I type a dot within a comment, I get a completion pop-up. Outside of comments a lone dot won't trigger a completion, and I get correct type based completions after other stuff.

<img width="542" alt="Screenshot 2023-08-01 at 12 55 15 PM" src="https://github.com/python-lsp/python-lsp-server/assets/10857610/2f139258-554b-404f-800b-471b463ab756">

Is this expected behavior? I tend to use complete sentences in my comments and this is leading to lots of comments accidentally ending in '.ArithmeticError' instead of '.' 😬

**Comment by ccordoba12** [comment]
Dots are trigger characters for completions in Python, as declared here:

https://github.com/python-lsp/python-lsp-server/blob/714a5829192133f5fd7813a099741c580338c0bd/pylsp/python_lsp.py#L245

And in this case completions are requested because you have a dot right next to a `#` (I guess if it's separated by a space, then you shouldn't see that).

However, I think you need to report this error to the eglot maintainers so that completions are not requested in comments.

**Comment by krassowski** [comment]
Counter-point: when `triggerKind` is `TriggerCharacter` server should ideally only return completions if the context makes sense (i.e. not after a dot occurring just after a comment).

**Comment by ccordoba12** [comment]
Sure, that's a good point. But my guess is that that is not the only scenario that triggers completions in eglot. I guess this should return them too:

```python
foo = 1

# foo.
```

@efroemling, could you check that?

**Comment by efroemling** [comment]
@ccordoba12 Here you go:
(apologies for the slow response; was out on vacation)

The 'foo' test above does give me the same global completion popup.
<img width="390" alt="Screenshot 2023-08-23 at 5 01 39 PM" src="https://github.com/python-lsp/python-lsp-server/assets/10857610/e1d85487-872e-4089-b2bb-55a50a329b27">

Perhaps interestingly, I'm also getting the same for simply a dot preceded by a space.
<img width="385" alt="Screenshot 2023-08-23 at 5 03 17 PM" src="https://github.com/python-lsp/python-lsp-server/assets/10857610/4de0862d-9190-478b-b441-77755e382db4">

Though I'm not getting completions for a dot (with or without spaces) without the comment.
<img width="331" alt="Screenshot 2023-08-23 at 5 04 55 PM" src="https://github.com/python-lsp/python-lsp-server/assets/10857610/2ed5a0a7-6b50-4ce8-8c2d-8ba0f98f8ed7">

Does this seem like it would fall under the python-lsp-server umbrella or should I ping the eglot folks?

**Comment by efroemling** [comment]
Just shaking the tree here, as this is still happening for me as of Emacs 29.3 with python-lsp-server 1.11.0 running through eglot.

TLDR: If I type `# This is a comment.` or even just `#.` I always get a completion pop-up for the period. That doesn't seem right.

Typing `This is not a comment.` does *not* give me a completion pop-up for the period unless 'comment' has been defined as a variable. That seems right.

Any other Emacs folks out there experiencing the same behavior? And is this something that should be fixed here or on the eglot end?
